### PR TITLE
 [MO] - [refactor] -> refactor clients

### PR DESCRIPTION
**Pull request by Seequick1**
Signed-off-by: Seequick1 <morsak@redhat.com>

### Type of change

- Refactoring

### Description

This PR refactor cluster clients with restructuring the test module. BaseITST was removed and the most important methods was moved to `KubeClusterResources`.  Now, we have only one Base for system tests and integration tests.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Seequick1** [comment]
strimzi-ci run tests test_only

**Comment by Seequick1** [comment]
strimzi-ci run tests test_only

**Comment by Seequick1** [comment]
@strimzi-ci run tests test_only

**Comment by Seequick1** [comment]
@strimzi-ci run tests test_only

**Comment by Seequick1** [comment]
@strimzi-ci run tests test_only testcase=OpenShiftTemplatesST

**Comment by Seequick1** [comment]
@strimzi-ci run tests test_only

**Comment by Seequick1** [comment]
@strimzi-ci run tests test_only

**Comment by Seequick1** [comment]
@strimzi-ci run tests test_only

**Comment by Seequick1** [comment]
@strimzi-ci run tests test_only

**Comment by Seequick1** [comment]
@strimzi-ci run tests test_only

File: api/src/test/java/io/strimzi/api/kafka/model/AbstractCrdIT.java [hunk 1]
```diff
@@ -7,16 +7,21 @@
...
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.jupiter.api.Assumptions.assumeTrue;
 
-public abstract class AbstractCrdIT extends BaseITST {
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public abstract class AbstractCrdIT {
+
+    protected KubeClusterResource cluster = KubeClusterResource.getInstance();
```
Comment (ppatierno) [inline api/src/test/java/io/strimzi/api/kafka/model/AbstractCrdIT.java]: the `AbstractResourceOperatorIT` class uses a @BeforeAll annotated method for getting this instance, why it's different here?

Comment (Seequick1) [reply api/src/test/java/io/strimzi/api/kafka/model/AbstractCrdIT.java]: It does not matter actually. I can rewrite it that on initialize phase.

Comment (Seequick1) [reply api/src/test/java/io/strimzi/api/kafka/model/AbstractCrdIT.java]: Or @Frawless can do it in following PR with refactor of resources :)

Comment (ppatierno) [reply api/src/test/java/io/strimzi/api/kafka/model/AbstractCrdIT.java]: I know that's the same, I was just wondering about the different approach and if it had any rationale for that.

**Pull request merged**